src/herder/HerderImpl.cpp

```diff
@@ -846,8 +846,25 @@ HerderImpl::triggerNextLedger(uint32_t ledgerSeqToTrigger)
     auto const& lcl = mLedgerManager.getLastClosedLedgerHeader();
     auto proposedSet = mTransactionQueue.toTxSet(lcl);
 
+    // We pick as next close time the current time unless it's before the last
+    // close time. We don't know how much time it will take to reach consensus
+    // so this is the most appropriate value to use as closeTime.
+    uint64_t nextCloseTime =
+        VirtualClock::to_time_t(mApp.getClock().system_now());
+    if (nextCloseTime <= lcl.header.scpValue.closeTime)
+    {
+        nextCloseTime = lcl.header.scpValue.closeTime + 1;
+    }
+
+    // Protocols including the "closetime change" (CAP-0034) externalize
```

Comment:
I don't think this is what CAP0034 says: we expect txsets to all be valid for the new close time.

This change implies that we expect them to be valid from `lcl.ct` up to `new_ct` (inclusive) which is a lot more restrictive.

Reply:
(Same question as to the previous comment.)

Reply:
(Same upcoming resolution.)